overlord/servicestate/quota_handlers.go

```diff
@@ -145,39 +146,15 @@ func quotaCreate(st *state.State, t *state.Task, action QuotaControlAction, allG
 
 func quotaCreateImpl(st *state.State, action QuotaControlAction, allGrps map[string]*quota.Group) (*quota.Group, map[string]*quota.Group, error) {
 	// make sure that the parent group exists if we are creating a sub-group
-	var grp *quota.Group
-	var err error
-	updatedGrps := []*quota.Group{}
+	var parentGrp *quota.Group
 	if action.ParentName != "" {
-		parentGrp, ok := allGrps[action.ParentName]
+		var ok bool
+		parentGrp, ok = allGrps[action.ParentName]
```

Comment:
why keep this check here instead of pushing it into internal.CreateQuota ?

Reply:
I think it makes for a more consistent set of signatures for the functions in internal,  also I think CreateQuotaInState can probably be reexported almost as is for the use case of mocking with just a bit more annoying for the caller